### PR TITLE
Packing reports performance

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Related to #4521 

Adds missing eager-loading in packing reports. Reduces total queries by ~76% and drastically reduces page load times (should be less than half the previous time).

#### What should we test?

Packing reports are working correctly.

#### Release notes

Improved performance in pacing reports.

Changelog Category: Changed

### CODE REVIEW

File: lib/open_food_network/packing_report.rb [hunk 1]
```diff
@@ -146,8 +147,8 @@ def is_by_customer?
       params[:report_type] == "pack_by_customer"
     end
 
-    def customer_code(email)
-      customer = Customer.where(email: email).first
+    def customer_code(order)
+      customer = order.customer
```
Comment (mkllnk) [inline lib/open_food_network/packing_report.rb]: This is fixing a bug as well. The report could have fetched the customer code given by another enterprise.

File: app/controllers/spree/admin/reports_controller_decorator.rb [hunk 2]
```diff
@@ -222,7 +222,8 @@ def my_suppliers
   end
 
   def suppliers_of_products_distributed_by(distributors)
-    distributors.map { |d| Spree::Product.in_distributor(d) }.flatten.map(&:supplier).uniq
+    distributors.map { |d| Spree::Product.in_distributor(d).includes(:supplier).all }.
```
Comment (mkllnk) [inline app/controllers/spree/admin/reports_controller_decorator.rb]: Definitely options for improvement here, but I agree it's out of scope. We probably want something like `Enterprise.is_primary_producer.supplying_variant_in(...)`.

**Comment by Matt-Yorkley** [comment]
I'm taking another look using live data from UK prod. It's pretty crazy..!

**Comment by Matt-Yorkley** [comment]
OK its basically crashing my laptop. I retested with production data and this reduced reports loading for one day from ~32 seconds to ~30 seconds. It also shows the huge difference we have in production, and the extent of other problems related to permissions when the amount of data is exponentially larger.

**Comment by Matt-Yorkley** [comment]
Retesting with production data; ~38% reduction in queries (from 1465 to 903), total page load reduced from 32.8 seconds to 30.2 seconds:

BEFORE:
Screenshot from 2019-11-30 11-51-40
AFTER:
Screenshot from 2019-11-30 12-15-59

**Comment by luisramos0** [comment]
great!
25th takes around that time 30secs.
the 25th and the 26th together take around 8mins for me.

**Comment by lin-d-hop** [comment]
This looks good. Tested at producer and food hub - Pack by Customer and Pack by Supplier.
Good to go.

**Pull request merged**